bnr-utils/nz_azConnector/nz_azConnector.go

```diff
@@ -301,9 +312,72 @@ func (cn *Conn) downloadBkp(outdir string, uniqueid string, blobpath string, str
     close(work)
     <- done
     log.Println("Total files downloaded:", filesdownloaded)
+    updateLocation(arrayLoc,outdir)
+    updateContents(arrayContents)
     return err
 }
 
+func updateLocation(arrLoc []string,outdir string){
+    for _,locFile := range arrLoc{
+        input, err := ioutil.ReadFile(locFile)
+        if err != nil {
+            log.Println("Unable to Override Contents and Location as download dir not Present.")
+            log.Fatalln(err)
+        }
+        lines := strings.Split(string(input), "\n")
+        if (len(lines) == 2 && !strings.HasSuffix(lines[len(lines) -2] , outdir)) {
+            f, err := os.OpenFile(locFile,
+                os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+            if err != nil {
+                log.Println("Unable to Override Location as Unable to open file.")
+                log.Fatalln(err)
+            }
+            defer f.Close()
+            textAppend := "1,1,1," + outdir + "\n"
+            if _, err := f.WriteString(textAppend); err != nil {
+                log.Println("Unable to Override Location as Write to File Failed.")
+                log.Fatalln(err)
+            log.Println("Location.txt override for cloud  backup")
+            }
+        }
+    }
+}
+
+func updateContents(arrContents []string){
+    for _,contentFile := range arrContents{
+        input, err := ioutil.ReadFile(contentFile)
+        if err != nil {
+            log.Println("Unable to Override Contents as unable to open contents.txt")
+            log.Fatalln(err)
+        }
+
+        lines := strings.Split(string(input), "\n")
+        var textline []string
+        var overide bool = false
+        for i := 0 ; i < len(lines) ; i++ {
+            line := lines[i]
+            token := strings.Split(line, ",")
+            if ( strings.HasSuffix(token[len(token)-1],"0" ) ) {
+                r := []rune(line)
+                str := string(r[:len(r)-1]) + "1"
+                textline = append(textline,str)
+                overide = true
+            } else {
+                textline = append(textline,line)
+            }
+        }

```

Comment:
Two questions

1. Are you trying to append "1" if the last character of `line` is `0` ? i.e. `...,10` and `....,0` both should be changed to `...,101` and `.....,01` ? Or are you trying to look for a `0` as the last comma separated number
2. In either case I suggest this - (I am going to assume you want the number ending with 0)

Also `override` isn't used.

```go
for _, line := range lines {
    if strings.HasSuffix(line, "0") {  // no need to split if its just suffix check
        line += "1"
    }
    textline = append(textline, line)
}
```

Reply:
I am looking for **",0"** to override with **",1"**  which we get only in cloud backup. i,e, backup taken on cloud.

Thats why need to split string and then compare **"0"** , changed condition from **strings.HasSuffix(token[len(token)-1],"0"** to 
**token[len(token)-1] == "0"** .

which will ensure that , only replace "0" with "1" and keep as it is the other values i,e, "1"..."2"...."10"...."100" etc.